locations/spiders/penske.py

```diff
@@ -1,114 +1,23 @@
-                        time_format="%I:%M %p",
-                    )
-            elif "," in day:
-                days = list(DAY_MAPPING.keys())
-                start_day, end_day = day.split(", ")
-                for i in days[days.index(start_day) : days.index(end_day) + 1]:
-                    opening_hours.add_range(
-                        day=DAY_MAPPING[i],
-                        open_time=start_time,
-                        close_time=end_time,
-                        time_format="%I:%M %p",
-                    )
-            else:
-                opening_hours.add_range(
-                    day=DAY_MAPPING[day],
-                    open_time=start_time,
-                    close_time=end_time,
-                    time_format="%I:%M %p",
-                )
-
-        return opening_hours.as_opening_hours()
-
-    def parse_store(self, response):
-        ref = re.search(r".+/(.+)$", response.url).group(1)
-
-        properties = {
-            "addr_full": response.xpath('//div[@id="location-left"]/p/text()').extract_first(),
-            "phone": response.xpath('//span[@itemprop="telephone"]/text()').extract_first(),
-            "city": response.xpath('//span[@itemprop="addressLocality"]/text()').extract_first(),
-            "state": response.xpath('//span[@itemprop="addressRegion"]/text()').extract_first(),
-            "postcode": response.xpath('//span[@itemprop="postalCode"]/text()').extract_first(),
-            "ref": ref,
-            "website": response.url,
-            "lat": float(response.xpath('//dt[@itemprop="latitude"]/text()').extract_first()),
-            "lon": float(response.xpath('//dt[@itemprop="longitude"]/text()').extract_first()),
-            "name": response.xpath('//h1[@itemprop="name"]/text()').extract_first(),
-        }
-
-        hours = self.parse_hours(response.xpath('//dl[@class="hours"]'))
-
-        if hours:
-            properties["opening_hours"] = hours
-
-        yield GeojsonPointItem(**properties)
-
-    def parse(self, response):
-        urls = response.xpath('//section[@class="locations-by-state"]/ul/li/a/@href').extract()
-        if urls:
-            urls.extend(response.xpath('//section[@class="locations-by-province"]/ul/li/a/@href').extract())
-
-        is_store = False
-
-        if not urls:
-            urls = response.xpath('//section[@class="locations-by-city"]/ul/li/a/@href').extract()
+    sitemap_urls = ["https://www.pensketruckrental.com/sitemap.xml"]
+    sitemap_rules = [(r"/locations/us/[-\w]+/[-\w]+/[0-9]+/$", "parse_sd")]
+    wanted_types = ["LocalBusiness"]
 
-        if not urls:
-            urls = response.xpath('//a[contains(@class,"location-link")]/@href').extract()
-            is_store = True
+    def sitemap_filter(self, entries):
+        yield from entries
 
-        for url in urls:
+    def post_process_item(self, item, response, ld_data):
+        item["ref"] = re.findall("[0-9]+", response.url)[0]
+        item.pop("email")
```

Comment:
`pop()` will throw an exception is the key doesn't exist. Add `None` default to prevent that exception.

```suggestion
        item.pop("email", None)
```